louve_addons/edi_purchase_diapar/wizard/invoice_supplier_price_update.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016-Today: Druidoo (<http://www.druidoo.net/>)
+# @author: Druidoo
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html
+
+from openerp import api, fields, models, _
+from openerp.exceptions import ValidationError
+
+supported_models = [
+    'account.invoice',
+]
+
+class InvoiceSupplierPriceUpdate(models.TransientModel):
+    _name = 'invoice.supplier.price.update'
+
+    show_discount = fields.Boolean()
+    partner_id = fields.Many2one(comodel_name='res.partner')
+    edi_line_ids = fields.One2many(
+        comodel_name='supplier.info.update.line',
+        inverse_name='inv_supplier_price_id'
+    )
+    edi_line2_ids = fields.One2many(
+        comodel_name='supplier.info.update.line',
+        inverse_name='inv_supplier_price_id'
+    )
+
+    @api.onchange('show_discount')
+    def _onchange_show_discount(self):
+        self.ensure_one()
+        if self.show_discount:
+            self.edi_line_ids = self.edi_line2_ids
+            self.edi_line2_ids = self.env['supplier.info.update.line']
+        else:
+            self.edi_line2_ids = self.edi_line_ids
+            self.edi_line_ids = self.env['supplier.info.update.line']
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super(InvoiceSupplierPriceUpdate, self).default_get(
+            fields_list=fields_list)
+        active_model = self._context.get('active_model', '')
+        active_id = self._context.get('active_id', False)
+        if active_model in supported_models and active_id:
+            active_obj = self.env[active_model].browse(active_id)
+            processed_lines = self.compute_process_lines(
+                active_model, active_obj)
+            show_discount = active_obj.partner_id.show_discount
+            res.update({
+                'edi_line_ids': processed_lines,
+                'edi_line_ids': processed_lines,
+                'partner_id': active_obj.partner_id.id,
+                'show_discount': show_discount,
+            })
+        return res
+
+    @api.model
+    def compute_process_lines(self, active_model, active_obj):
+        """
+        update product prices/order line prices based on latest EDI supplier prices (Only for EDI suppliers)
+        :param active_model: Current active model
+        :param active_obj: Current active object
+        :return: computed lines
+        """
+        lines = []
+        partner_id = active_obj.partner_id
+        obj_lines = 'order_line' in active_obj and active_obj.order_line \
+                    or active_obj.invoice_line_ids
+        # For invoice date
+        active_id = self._context.get('active_id', False)
+        invoice_id = self.env['account.invoice'].browse(active_id)
+        supplier_price_list_obj = self.env['supplier.price.list']
+        invoice_date = False
+        if invoice_id:
+            invoice_date = invoice_id[0].date_invoice
```

Comment:
For instance, by cleaning up I mean:

```suggestion
        price_date = active_model == 'account.invoice' \
            and active_obj.date_invoice or False
```